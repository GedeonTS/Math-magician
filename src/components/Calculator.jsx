import React, { useState, useEffect } from "react";
import calculate from "../logic/calculate";


const Calculator = () => {
  const [state, setState] = useState({
    total: null,
    next: null,
    operation: null,
  });

  
  useEffect =() => {
    setState({
      total: null,
      next: null,
      opreration: null,
    });
     
  };



  const updateObject = (e) => {
    const myObj = calculate(state, e.target.textContent);
    setState(myObj);
  };

  const { next, operation, total } = state;
  const op = operation === "%" ? "mod" : operation;
  let result = "";
  if (total) {
    result = `${total} ${op || ""} ${next || ""}`;
  } else if (next) {
    result = `${next} ${op || ""}`;
  }

  return (
    <div className="container">
      <div className="input-field">
        <p>{result || 0}</p>
      </div>
      <div className="wrapper">
        <div className="row">
          <button
            type="button"
            onClick={updateObject}
            className="clear-small-button"
          >
            AC
          </button>
          <button
            type="button"
            onClick={updateObject}
            className="clear-small-button"
          >
            +/-
          </button>
          <button
            type="button"
            onClick={updateObject}
            className="clear-small-button"
          >
            %
          </button>
          <button
            type="button"
            onClick={updateObject}
            className="clear-small-button operator-button"
          >
            รท
          </button>
        </div>
        <div className="row">
          <button
            type="button"
            onClick={updateObject}
            className="clear-small-button"
          >
            7
          </button>
          <button
            type="button"
            onClick={updateObject}
            className="clear-small-button"
          >
            8
          </button>
          <button
            type="button"
            onClick={updateObject}
            className="clear-small-button"
          >
            9
          </button>
          <button
            type="button"
            onClick={updateObject}
            className="clear-small-button operator-button"
          >
            x
          </button>
        </div>
        <div className="row">
          <button
            type="button"
            onClick={updateObject}
            className="clear-small-button"
          >
            4
          </button>
          <button
            type="button"
            onClick={updateObject}
            className="clear-small-button"
          >
            5
          </button>
          <button
            type="button"
            onClick={updateObject}
            className="clear-small-button"
          >
            6
          </button>
          <button
            type="button"
            onClick={updateObject}
            className="clear-small-button operator-button"
          >
            -
          </button>
        </div>
        <div className="row">
          <button
            type="button"
            onClick={updateObject}
            className="clear-small-button"
          >
            1
          </button>
          <button
            type="button"
            onClick={updateObject}
            className="clear-small-button"
          >
            2
          </button>
          <button
            type="button"
            onClick={updateObject}
            className="clear-small-button"
          >
            3
          </button>
          <button
            type="button"
            onClick={updateObject}
            className="clear-small-button operator-button"
          >
            +
          </button>
        </div>
        <div className="row">
          <button
            type="button"
            onClick={updateObject}
            className="clear-large-button"
          >
            0
          </button>
          <button
            type="button"
            onClick={updateObject}
            className="clear-small-button"
          >
            .
          </button>
          <button
            type="button"
            onClick={updateObject}
            className="clear-small-button operator-button"
          >
            =
          </button>
        </div>
      </div>
    </div>
  );
};
export default Calculator;
