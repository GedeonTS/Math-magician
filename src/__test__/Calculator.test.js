import React from "react";
import renderer from 'react-test-renderer';
import Calculator from "../components/Calculator";
import '@testing-library/jest-dom';

describe('Test of Calculator page', () => {
    it('test if Calculator page is rendered', () => {
        const calculatorTest = renderer
            .create(<Calculator />)
            .toJSON();
        expect(calculatorTest).toMatchSnapshot();
    });
});
