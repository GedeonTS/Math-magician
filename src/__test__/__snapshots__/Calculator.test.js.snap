// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test of Calculator page test if Calculator page is rendered 1`] = `
<section
  className="calculator-section"
>
  <h2>
    Let's do some Maths!
  </h2>
  <div
    className="container"
  >
    <div
      className="input-field"
    >
      <p>
        0
      </p>
    </div>
    <div
      className="wrapper"
    >
      <div
        className="row"
      >
        <button
          className="clear-small-button"
          onClick={[Function]}
          type="button"
        >
          AC
        </button>
        <button
          className="clear-small-button"
          onClick={[Function]}
          type="button"
        >
          +/-
        </button>
        <button
          className="clear-small-button"
          onClick={[Function]}
          type="button"
        >
          %
        </button>
        <button
          className="clear-small-button operator-button"
          onClick={[Function]}
          type="button"
        >
          รท
        </button>
      </div>
      <div
        className="row"
      >
        <button
          className="clear-small-button"
          onClick={[Function]}
          type="button"
        >
          7
        </button>
        <button
          className="clear-small-button"
          onClick={[Function]}
          type="button"
        >
          8
        </button>
        <button
          className="clear-small-button"
          onClick={[Function]}
          type="button"
        >
          9
        </button>
        <button
          className="clear-small-button operator-button"
          onClick={[Function]}
          type="button"
        >
          x
        </button>
      </div>
      <div
        className="row"
      >
        <button
          className="clear-small-button"
          onClick={[Function]}
          type="button"
        >
          4
        </button>
        <button
          className="clear-small-button"
          onClick={[Function]}
          type="button"
        >
          5
        </button>
        <button
          className="clear-small-button"
          onClick={[Function]}
          type="button"
        >
          6
        </button>
        <button
          className="clear-small-button operator-button"
          onClick={[Function]}
          type="button"
        >
          -
        </button>
      </div>
      <div
        className="row"
      >
        <button
          className="clear-small-button"
          onClick={[Function]}
          type="button"
        >
          1
        </button>
        <button
          className="clear-small-button"
          onClick={[Function]}
          type="button"
        >
          2
        </button>
        <button
          className="clear-small-button"
          onClick={[Function]}
          type="button"
        >
          3
        </button>
        <button
          className="clear-small-button operator-button"
          onClick={[Function]}
          type="button"
        >
          +
        </button>
      </div>
      <div
        className="row"
      >
        <button
          className="clear-large-button"
          onClick={[Function]}
          type="button"
        >
          0
        </button>
        <button
          className="clear-small-button"
          onClick={[Function]}
          type="button"
        >
          .
        </button>
        <button
          className="clear-small-button operator-button"
          onClick={[Function]}
          type="button"
        >
          =
        </button>
      </div>
    </div>
  </div>
</section>
`;
